torch/nn/parallel/distributed.py

```diff
@@ -380,14 +380,16 @@ def forward(self, *inputs, **kwargs):
         else:
             output = self.module(*inputs, **kwargs)
 
-        # We'll return the output object verbatim since it is a freeform object.
-        # We need to find any tensors in this object, though, because we need to
-        # figure out which parameters were used during this forward pass,
-        # to ensure we short circuit reduction for any unused parameters.
-        if self.find_unused_parameters:
-            self.reducer.prepare_for_backward(list(_find_tensors(output)))
-        else:
-            self.reducer.prepare_for_backward([])
+        if self.training and torch.is_grad_enabled():
```

Comment:
self.training has nothing to do with whether gradients are called or not. It's only used for whether BatchNorm and Dropout are in training or eval mode.
Dont check for this as it isn't hard indicator of training or eval mode (only a hint)

Reply:
Removed the `self.training` check.